components/button/test/button.test.js

```diff
@@ -47,16 +47,20 @@ describe('button', () => {
 			let submitted = false;
 			const addedElements = addElement(
 				textToDomToParent(
-					`<form onsubmit="return false" name="testForm" id="testForm"><${COMPONENT_NAME}>Button Text</${COMPONENT_NAME}></form>`
+					`<form onsubmit="return false" id="testForm"><${COMPONENT_NAME} name="testForm" value="testValue">Button Text</${COMPONENT_NAME}></form>`
 				)
 			);
 			await waitNextTask();
 			const formElement = addedElements[0];
 			const actualElement = formElement.firstChild;
-			formElement.addEventListener('submit', () => (submitted = true));
+			formElement.addEventListener('submit', ({ submitter }) => {
+				submitted =
```

Comment:
You could have done something like:
`expect(submitter === actualElement.querySelector('button')).to.equal(true)`;
This will also make sure they are the same, no?
All in all there are 3 things tested:
1. That the form is submitted
2. That if we add name and type they are reflected
3. That the submitter is the hidden button
These should be in 3 different `it`s.

Reply:
i was following your verbal recommendation :)

i would definitely prefer using expect in the way you offered. however, since i'm not the original author of this test, and it was already reviewed and approved this way, i'd rather not refactor anything.

1. if the form is submitted without the relevant name and value passed, there is no significance to the submission.
2. it is checked - both are set on 'vwc-button''s setup.
3. i'm not sure there's significance in verifying that the submitter is the hidden button - that's an implementation detail which might change (for instance - input instead of button etc.).

Reply:
notice `submitter` compatibility. how will it pass in Safari?
![image](https://user-images.githubusercontent.com/15140652/135578701-b00f2ecf-8516-47db-83c3-2bfa353cf3aa.png)

Reply:
good catch! we need to rethink of ways to test this one then